engine/crates/validation/src/validate/objects.rs

```diff
@@ -13,6 +13,10 @@ pub(crate) fn validate_object<'a>(
     });
 
     ctx.with_fields(parent_type_name, &obj.fields, |ctx, fields| {
+        if fields.is_empty() {
+            diagnostics::empty_object(parent_type_name, ctx)
+        }
+
```

Comment:
Sorry, nitpicking this quite a lot, but should we be doing the same thing for interfaces & input objects?

Reply:
done & pushed